server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -363,10 +363,11 @@ protected ReplicationOperation.Replicas<BulkShardRequest> primaryTermValidationR
     /**
      * This {@link org.opensearch.action.support.replication.TransportReplicationAction.ReplicasProxy} implementation is
      * used for primary term validation and is only relevant for TransportShardBulkAction replication action.
-     *
+     * <p>
+     * Visible for tests
      * @opensearch.internal
      */
-    private final class PrimaryTermValidationProxy extends WriteActionReplicasProxy {
+    public final class PrimaryTermValidationProxy extends WriteActionReplicasProxy {
```

Comment:
lets revert this as the class is not being used anywhere else apart from this class.

Reply:
Addressed